### ISSUE TITLE
피드백 - 윤창식 튜터

### ISSUE THREAD

**Issue by CaesiumY** [HuaHuaChiChi/nbc_week3_todolist#3]
# 개요
컴포넌트를 나누는 게 아직은 익숙하지 않으신 거 같아요... 큰 단위의 컴포넌트만 만들어주시고 props도 엄청 많이 넘겨주셨네요. 다음엔 컴포넌트를 나누는 기준을 잘 세워서 좀 더 기능적으로 분배된 코드를 볼 수 있으면 좋겠습니다! 

# 리뷰
## 빌드 결과물?
![image](https://github.com/HuaHuaChiChi/nbc_week3_todolist/assets/35517199/58a46d3d-be39-411f-b857-c3280128af72)
- 혹시 여기 `static` 폴더와 루트의 `index.html`은 무슨 용도일까요? 제 추측으로 미루어보건대 빌드한 결과물일까요? 만약 의도한 바라면 빌드 결과물은 깃허브에 올리지 않으셔도 됩니다!

## 컴포넌트 세분화
![image](https://github.com/HuaHuaChiChi/nbc_week3_todolist/assets/35517199/2f48fca5-f173-44aa-aa06-a8e73683e65c)
- 컴포넌트를 너무 크게 생각하시고 나누신 거 같아요ㅠ 컴포넌트의 목적인 재사용성과 가독성, 유지 보수가 제대로 작동하지 않는 거 같습니다... 머릿속으로 또는 그림으로 그려서라도 먼저 컴포넌트를 설계하시고 나서 분리해보세요! 제가 진행했던 실시간 세션을 다시 보는 것도 도움이 될 겁니다. 초반에 스윽 스윽 나눠버리거든요😅 그 패턴을 암기하셔도 좋습니다!

## CSS 선택자 이름
https://github.com/HuaHuaChiChi/nbc_week3_todolist/blob/6c9718796837d92452e2860167f55a34b96794ae/src/App.css#L49
- CSS 선택자는 일관성을 위해 `kebab-case`를 쓰시는 것이 좋습니다! `border-radius` 같은 속성을 보시면 아시다시피 CSS 파일은 kebab-case 표기법을 따르고 있거든요. 

## 객체 키 값 단축하기
https://github.com/HuaHuaChiChi/nbc_week3_todolist/blob/6c9718796837d92452e2860167f55a34b96794ae/src/components/ToDoListContainer.jsx#L14-L19
- 객체에서 키와 값이 같은 이름이면 하나로 단축할 수 있습니다! `toDoTitle: toDoTitle,` 이게 아니라 `toDoTitle,` 로 단축할 수 있어요!
- `id`를 길이 값으로 하면 중복이 생겨 오류가 날 수 있어요. 예를 들어 길이에 따라 만들어진 배열이 있다면 1 2 3 4 라는 아이디를 가지고 있을 거예요. 그러다가 2가 삭제되면 1 3 4만 남겠죠? 다음 아이템이 들어오면 현재 길이에 +1 된 아이디를 가지기 때문에 1 3 4 4의 아이디를 가진 리스트가 생성됩니다... 그럼 아이디는 유니크 값이 아니게 되고 오류가 생기게 되겠죠. 대신에 실시간 강의 때 말씀드린 `crypto.randomUUID()` 나 더 간단한 `Date.now()` 를 사용하시는 게 더 좋습니다!

## 네이버에서 찾아보기
https://github.com/HuaHuaChiChi/nbc_week3_todolist/blob/6c9718796837d92452e2860167f55a34b96794ae/src/components/ToDoListContainer.jsx#L28-L35
- 인터넷에서 찾아봐도 괜찮습니다! 안 그러는 개발자는 존재하지 않거든요. 다만 본인의 것으로 만들 수 있도록 스스로 따라 쳐보거나 이해하도록 하는 노력이 필요합니다. 안 괜찮은 건 네이버에서 찾았다는 거... 정도? 구글에서 검색하는 게 좀 더 찾기 쉽거든요!

# 마무리
처음에는 컴포넌트 분리한 모습을 보고 다 구현 못했을 줄 알았는데 제 예상과는 반대로 구현은 잘 해주셨어요! 다만 컴포넌트를 너무 큰 조각으로 나눈 점이 좀 아쉽네요. 컴포넌트 분리를 먼저 숙고해보기 전에 손이 먼저 나가서 코드를 작성하셔서 그 거 같아요. 구현은 잘 해주셨으니 앞으로 더 많은 코드를 작성하시면서 익숙해지시면 될 것 같습니다! 코드를 좋게 작성하려다 구현 못 하는 것 보다는 코드를 하나에 몰아서 작성해버리더라도 제대로 동작하는 프로덕트를 만드는 게 회사에서 원하는 개발자 상입니다!

**Comment by HuaHuaChiChi** [comment]
네 강의를 듣고 props와 컴포넌트 분리가 아직 어려워서, STATE는 이해 했으니 구현부터 하고 나중에 나눠보자!!
라는 마음으로 진행을 했는데 나중에 부랴부랴 나누려니 쉽지 않아서 코드가 두동강 나고 말았네요,,,

빌드한 파일은 리액트 파일을 깃허브로 배포해보고 싶어서 이것저것 해보다가 성공했는데 이건 잘못된 방식이였군요,,!!,,

스탠다드반 과제가 마침 todolist를 다시 구현하는 것 이라서 창식튜터님의 해설을 다시 보고 참고하면서 진행하고자 합니다.
피드백 감사합니다!!

_네이버는 좀 짜치긴 하네요 ㅋㅋㅋㅋㅋㅋㅋㅋㅋ_

**Comment by CaesiumY** [comment]
@HuaHuaChiChi 깃허브 페이지 배포라면 https://github.com/tschaub/gh-pages?tab=readme-ov-file#deploying-with-github-actions-and-a-named-script 여기 보시면 gh-pages 라는 패키지를 설치해서, deploy 명령어를 설정하면 쉽게 배포가 가능합니다!
